ethereum/contracts/TablelandRigs.sol

```diff
@@ -446,11 +464,25 @@ contract TablelandRigs is
     function trainRig(uint256 tokenId) public whenNotPaused {
         // Check the Rig `tokenId` exists
         if (!_exists(tokenId)) revert OwnerQueryForNonexistentToken();
+
         // Verify `msg.sender` is authorized to train the specified Rig
-        if (ownerOf(tokenId) != _msgSenderERC721A())
-            revert ITablelandRigPilots.Unauthorized();
+        address sender = _msgSenderERC721A();
+        address tokenOwner = ownerOf(tokenId);
+
+        if (
+            _delegateCash.checkDelegateForToken(
+                sender,
+                tokenOwner,
+                address(this),
+                tokenId
+            )
+        ) {
+            sender = tokenOwner;
+        }
 
-        _pilots.trainRig(_msgSenderERC721A(), tokenId);
+        if (tokenOwner != sender) revert ITablelandRigPilots.Unauthorized();
```

Comment:
this could be a function/modifier to save on contract size since it's reused three times